src/components/ReceiverInput.tsx

```diff
@@ -0,0 +1,137 @@
+import { INCORRECT_FORMAT, GENERIC } from '../constants';
+import { TransactionActionCreators } from '../actions/transactionActions';
+
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import { useTransactionContext, useUpdateTransactionContext } from '../contexts/TransactionContext';
+import useApiBalance from '../hooks/useApiBalance';
+import useBalance from '../hooks/useBalance';
+import usePrevious from '../hooks/usePrevious';
+
+import useReceiver from '../hooks/useReceiver';
+import AccountIdenticon from './AccountIdenticon';
+
+interface Props {
+  setError: (value: string) => void;
+}
+
+const useStyles = makeStyles(() => ({
+  container: {
+    padding: '0',
+    float: 'left'
+  },
+  row: {
+    border: '1px solid grey',
+    borderRadius: '5px 5px 0 0',
+    display: 'flex',
+    padding: '10px 20px',
+    alignItems: 'center'
+  },
+  address: {
+    marginLeft: '10px'
+  },
+  balance: {
+    marginLeft: 'auto'
+  },
+  input: {
+    marginLeft: '10px',
+    width: '80%'
+  }
+}));
+
+function ReceiverInput({ setError }: Props) {
+  const classes = useStyles();
+  const [addressInput, setAddresInput] = useState('');
+  const [formatFound, setFormatFound] = useState('');
+  const [showBalance, setShowBalance] = useState(false);
+
+  const { setReceiver, validateAccount } = useReceiver();
+
+  const { dispatchTransaction } = useUpdateTransactionContext();
+  const { receiverAddress } = useTransactionContext();
+  const { api, address } = useApiBalance(addressInput, formatFound, false);
+  const state = useBalance(api, address, true);
+
+  const {
+    targetChainDetails: { targetChain }
+  } = useSourceTarget();
+  const prevTargetChain = usePrevious(targetChain);
+
+  const reset = useCallback(() => {
+    dispatchTransaction(TransactionActionCreators.setGenericAccount(null));
+    dispatchTransaction(TransactionActionCreators.setDerivedAccount(null));
+    setShowBalance(false);
+    setError('');
+  }, [dispatchTransaction, setError]);
+
+  useEffect(() => {
+    if (prevTargetChain !== targetChain) {
+      reset();
+      setAddresInput('');
+    }
+  }, [addressInput, prevTargetChain, receiverAddress, reset, targetChain]);
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const receiver = event.target.value;
+    reset();
+    setAddresInput(receiver);
+    if (!receiver) {
+      return;
+    }
+
+    const { formattedAccount, formatFound } = validateAccount(receiver)!;
+    setFormatFound(formatFound);
+    if (formatFound === INCORRECT_FORMAT) {
+      setError('Invalid address.');
+      return;
+    }
+
+    if (formatFound === GENERIC) {
+      dispatchTransaction(TransactionActionCreators.setGenericAccount(receiver));
+      return;
+    }
+
+    if (formatFound === targetChain) {
+      setReceiver(formattedAccount);
+    } else {
+      dispatchTransaction(TransactionActionCreators.setDerivedAccount(formattedAccount));
```

Comment:
We shouldn't be assuming that it's sourceChain account if `formatFound` is neither `GENERIC` nor `targetChain`.

Reply:
why not? what are the other options then?

Reply:
Yes, there is 16k possible SS58 prefix values, and we should only handle 3 of them: `42 (generic)`, source, target.

Reply:
Still something to fix.

Reply:
Done.